### ISSUE TITLE
"New key association request" every time I connect

### ISSUE THREAD

**Issue by daishi** [keepassxreboot/keepassxc#2552]
## Expected Behavior

Every time that I unlock and reconnect to KeePassXC from Firefox using KeePassXC-Browser, I need to create a name through the "New key association request" dialog. This appears to save some state, because it tells me that I already have a association with that name (since I use the same name each time), but it doesn't seem to ever use the saved association.

Based on https://keepassxc.org/quickstart/ and the application behavior I believe that I'm only supposed to need to do this once. Version information below. Let me know if there's additional information I can provide to help clarify my situation.

## Debug/Version Info

KeePassXC-Browser Version: 1.3.0
KeePassXC Version: 2.3.1
Revision: 2fcaeea
Firefox 63.0.3 on Ubuntu 18.04

**Comment by droidmonkey** [comment]
This indicates that your browser is forgetting the connection. KeePassXC is saving it in the database correctly.

**Comment by daishi** [comment]
Hi, Thanks for the reply. I believe that I'm using the default versions of both Firefox and KeePassXC as provided in the Ubuntu repos:
```
$ apt-cache policy keepassxc
keepassxc:
  Installed: 2.3.1+dfsg.1-1
  Candidate: 2.3.1+dfsg.1-1
  Version table:
 *** 2.3.1+dfsg.1-1 500
        500 http://us.archive.ubuntu.com/ubuntu bionic/universe amd64 Packages
        100 /var/lib/dpkg/status
$ apt-cache policy firefox
firefox:
  Installed: 64.0+build3-0ubuntu0.18.04.1
  Candidate: 64.0+build3-0ubuntu0.18.04.1
  Version table:
 *** 64.0+build3-0ubuntu0.18.04.1 500
        500 http://us.archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages
        500 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages
        100 /var/lib/dpkg/status
     59.0.2+build1-0ubuntu1 500
        500 http://us.archive.ubuntu.com/ubuntu bionic/main amd64 Packages
```
If I go into the KeePassXC-Browser settings "Connected Databases" panel in Firefox I do see my connection, but when I click on the KeePassXC-Browser icon in the browser toolbar, I see the message

> KeePassXC-Browser has not been configured. Press the connect button to pair with KeePassXC.

(This is regardless of whether the KeePassXC application itself is locked or unlocked). Do you have thoughts on what I might examine next?

**Comment by droidmonkey** [comment]
Ah please upgrade to 2.3.4 to correct this issue. You can use the AppImage or ppa.

**Comment by daishi** [comment]
Thanks that resolved my situation. Related, I believe that these associations are stored in the .kdbx file itself. Is there a way to review these associations? I see in the Tools->Settings->Browser Integration that there are buttons to "Disconnect all browsers" and to "Forget all remembered permissions" but nothing to list/manage the permissions. Have I missed something? If I want to ensure that I understand the connection permissions that I've granted, should I "Forget all" and re-associate?

**Comment by daishi** [comment]
(Closing since the main issue was resolved by the update to 2.3.4).
